asset-transfer-private-data/chaincode-go/go/privateAssetTransfer.go

```diff
@@ -0,0 +1,503 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// Asset describes basic details that are visible to all organizations
+type Asset struct {
+	ObjectType 		 string `json:"object_type"`  //docType is used to distinguish the various types of objects in state database
+	ID             string `json:"asset_id"`
+	Color          string `json:"color"`
+	Size           int    `json:"size"`
+	Owner          string `json:"owner"`
+}
+
+// AssetPrivateDetails describes details that are private to owners
+type AssetPrivateDetails struct {
+	ID             string `json:"ID"`
+	AppraisedValue int    `json:"appraisedValue"`
+}
+
+type SmartContract struct {
+	contractapi.Contract
+}
+
+
+// create a new asset by placing asset public details in the "assetCollection",
```

Comment:
Go function comments should start with the function name... 
//CreateAsset creates a new asset...

The details are not really public, they are still in a private collection...